### PR TITLE
Replace create with join room

### PR DESCRIPTION
**Pull request by MattiaPrimavera**
This PR allowed to test joining a call with 2 participants locally:

- Replace `createRoom` call with `joinRoom` only, since it now takes care of creating the room if it does not exist yet, and if the user has the rights to
- fix logout button issue (corresponding action was not being dispatched)
- set functions types back to `./functions/src/types` in order to avoid `tsc` transpiration issues (due to the use of relative paths to access the types declared in `<rootProject>/types` as `../../types` from `firebase functions`
- Email/password login with emulator
- Remove anonymous login by default when using emulator, add a button to login anonymously instead

I used a basic UI to interact with the emulator and simulate user logins

Screenshot 2020-12-11 at 17 03 26

### To review
- review by commit (no need to review the first one, as I only moved back the functions `types`, where they were originally

### CODE REVIEW

File: src/pages/PremiumVideoCall/components/PreJoinScreens/DeviceSelectionScreen/DeviceSelectionScreen.tsx [hunk 1]
```diff
@@ -82,24 +83,17 @@ export default function DeviceSelectionScreen({
...
-            .then((response) => response.roomId)
-            .then((roomId) => {
-                return api.joinRoom(roomId)
-            })
-            .then((response) => {
-                const { jwtAccessToken } = response
-                return jwtAccessToken
-            })
-            .then((jwtAccessToken) => connect(jwtAccessToken))
+        const response = await api.joinRoom(roomId, 'test-password') // TODO pass the password variable
```
Comment (HugoGresse) [inline src/pages/PremiumVideoCall/components/PreJoinScreens/DeviceSelectionScreen/DeviceSelectionScreen.tsx:88]:
```suggestion
        const { jwtAccessToken } = await api.joinRoom(roomId, 'test-password') // TODO pass the password variable
```

File: src/pages/PremiumVideoCall/components/PreJoinScreens/DeviceSelectionScreen/DeviceSelectionScreen.tsx [hunk 2]
```diff
@@ -82,24 +83,17 @@ export default function DeviceSelectionScreen({
...
+        const { jwtAccessToken } = response
```
Comment (HugoGresse) [inline src/pages/PremiumVideoCall/components/PreJoinScreens/DeviceSelectionScreen/DeviceSelectionScreen.tsx:89]:
```suggestion
```

File: src/pages/PremiumVideoCall/state/usePasscodeAuth/usePasscodeAuth.ts [hunk 3]
```diff
@@ -17,26 +18,15 @@ export function getPasscode() {
...
-            return response.roomId
-        })
-        .then((roomId) => {
-            return api.joinRoom(roomId)
-        })
-        .then((response) => {
-            const { jwtAccessToken } = response
-            return jwtAccessToken
-        })
+    const response = await api.joinRoom(roomId, password)
```
Comment (HugoGresse) [inline src/pages/PremiumVideoCall/state/usePasscodeAuth/usePasscodeAuth.ts:27]:
```suggestion
    const  { jwtAccessToken } = await api.joinRoom(roomId, password)
```

File: src/pages/PremiumVideoCall/state/usePasscodeAuth/usePasscodeAuth.ts [hunk 4]
```diff
@@ -17,26 +18,15 @@ export function getPasscode() {
...
+    const { jwtAccessToken } = response
```
Comment (HugoGresse) [inline src/pages/PremiumVideoCall/state/usePasscodeAuth/usePasscodeAuth.ts:28]:
```suggestion
```

File: src/services/api.ts [hunk 5]
```diff
@@ -12,19 +12,11 @@ export class Api {
         this.jwtToken = jwtToken
     }
 
-    async createRoom(): Promise<NewRoomResponse> {
```
Comment (HugoGresse) [inline src/services/api.ts:15]: we may keep it for users wanting to plan a meeting in advance

**Review by ntchou — APPROVED** [review]

**Pull request merged**